src/AuthenticationPortal.Web/Middlewares/CustomExceptionMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Authentication
+{
+    // You may need to install the Microsoft.AspNetCore.Http.Abstractions package into your project
+
+    public class CustomExceptionMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public CustomExceptionMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next.Invoke(context);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionAsync(context, ex);
+            }
+        }
+
+        private static Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            var response = context.Response;
+            var customException = exception.InnerException as BaseCustomException;
+            var statusCode = (int)HttpStatusCode.InternalServerError;
+            var message = "Unexpected Error";
+
+            if (null != customException)
+            {
+                message = customException.Message;
+                statusCode = customException.Code;
+            }
+
+            response.ContentType = "application/json";
+            response.StatusCode = statusCode;
+            return response.WriteAsync(JsonConvert.SerializeObject(new CustomErrorResponse
+            {
+                Code = statusCode,
```

Comment:
The code you return as a part of response could never be http status code.. 
You need to define your own custom error codes 
For example,
If we have an error message which says,
“Unexpected system error occurred. Please Try again after some time.”
This message will have an unique error code say error code: 1.
So the api consumer will be aware, that  error code 1  means the  default internal server  error response is returned by your api ( he may or may not take any action for this error code or he may map the error code to some other error code and message he wants to show on ui) 
Will explain this in detail if needed